pkg/gardenlet/controller/backupentry/add.go

```diff
@@ -138,3 +139,24 @@ func getBackupBucketLastOperation(obj client.Object) *gardencorev1beta1.LastOper
 
 	return backupBucket.Status.LastOperation
 }
+
+// restoreAfterMigratePredicate returns a predicate which returns false if the core.gardener.cloud/v1beta1.BackupEntry has been
+// successfully migrated and does not have the `gardener.cloud/operation: restore` annotation.
+// It returns true in all other cases.
+func restoreAfterMigratePredicate() func(obj client.Object) bool {
```

Comment:
Do we need to return the function?
```suggestion
func restoreAfterMigratePredicate(obj client.Object) bool {
```